src/components/pages/profile/EditEmail/EditEmail.jsx

```diff
@@ -0,0 +1,119 @@
+import React, { createRef, useState } from 'react'
+
+import HeaderContainer from '../../../layout/Header/HeaderContainer'
+import EditPasswordField from '../EditPassword/EditPasswordField/EditPasswordField'
+import PersonalInformationsField from '../PersonalInformations/PersonalInformationsField/PersonalInformationsField'
+import { checkIfEmailIsValid } from '../../create-account/domain/checkIfEmailIsValid'
+import { updateEmail } from '../repository/updateEmail'
+import { toast } from 'react-toastify'
+import PropTypes from 'prop-types'
+
+const EditEmail = ({ redirectToPersonnalInformationPage }) => {
+  const [email, setEmail] = useState('')
+  const [password, setPassword] = useState('')
+  const [emailErrors, setEmailErrors] = useState(null)
+  const [passwordErrors, setPasswordErrors] = useState()
+
+  let _handleSubmit = function () {
+    return event => {
+      event.preventDefault()
+      const payload = {
+        new_email: email,
+        password: password,
+      }
+
+      return updateEmail(payload)
+        .then(async result => {
+          if (result.status === 204) {
+            setPasswordErrors(null)
+            toast.success('L’e-mail a bien été envoyé.')
+            return redirectToPersonnalInformationPage()
+          }
+
+          const answer = await result.json()
+
+          if (answer.password) setPasswordErrors(answer.password)
+        })
+        .catch(() => {
+          toast.error('La modification de l’adresse e-mail a échouée.')
+        })
+    }
+  }
+
+  let _handleEmailChange = function () {
+    return event => {
+      const email = event.target.value
+      setEmail(email)
+
+      if (!checkIfEmailIsValid(email)) {
+        setEmailErrors(["Format de l'e-mail incorrect"])
+      } else {
+        setEmailErrors(null)
+      }
+    }
+  }
+
+  let _handlePasswordChange = function () {
+    return event => setPassword(event.target.value)
+  }
```

Comment:
Je comprends pas trop la syntaxe ici, on ne pourrait pas plutôt faire:

```suggestion
  const _handlePasswordChange = event => setPassword(event.target.value)
```

Et plus bas:
```
onChange={_handlePasswordChange}
```
 ?
 
 Idem pour le mail